core/settlement/engine.go

```diff
@@ -120,12 +121,8 @@ func (e *Engine) Update(positions []events.MarketPosition) {
 	e.mu.Lock()
 	for _, p := range positions {
 		party := p.Party()
-		if ps, ok := e.pos[party]; ok {
-			// ATM, this can't possibly return an error, hence we're ignoring it
-			_ = ps.update(p)
-		} else {
-			e.pos[party] = newPos(p)
-		}
+		e.settledPosition[party] = p.Size()
+		e.lastMarkPrice = p.Price()
```

Comment:
Not sure about this. The last Mark price should come from the market/snapshot IMO. As far as settlement engine is concerned, we have parties with a certain size/position, and some unsettled trades. All positions at MTM are held at the same price. When we call this function, can we really be sure that the price from the market position event is the right one?

Reply:
this has nothing to do with the snapshot. it's not used by the snapshot, only by silly testing code.

Reply:
I know this doens't have anything to do with the snapshot stuff, but what is the reason we set `lastMarkPrice` in the loop here? solely for testing, or is it something that is used elsewhere?

Reply:
well of course. it's artificially initialising settled position out of thin air so it must also initialise the last mark price because out side of this test scenario, it is impossible to have a settled position without having gone through mark to market which would have set lastMarkPrice.